File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(p, s) {
    let answer = [];
    let day = 0;
    let cnt = 0;

    while (p.length > 0) {
        if (p[0] + day * s[0] >= 100) {
            s.shift();
            p.shift();
            cnt++;
        } else {
            day++;
            if (cnt > 0) {
                answer.push(cnt);
                cnt = 0;
            }
        }
    }
    answer.push(cnt);
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(p, s):
    answer = []
    day = 0
    cnt = 0
    while len(p)> 0:
        if p[0]+day*s[0]>=100:
            s.pop(0)
            p.pop(0)
            cnt+=1
        else:
            day+=1
            if cnt > 0:
                answer.append(cnt)
                cnt = 0
                
    answer.append(cnt)
    return answer